graylog2-web-interface/src/views/components/SearchBar.jsx

```diff
@@ -21,90 +23,100 @@ import HorizontalSpacer from 'views/components/horizontalspacer/HorizontalSpacer
 import { QueriesActions } from 'views/stores/QueriesStore';
 import { CurrentQueryStore } from 'views/stores/CurrentQueryStore';
 import { StreamsStore } from 'views/stores/StreamsStore';
-import { QueryFiltersActions, QueryFiltersStore } from 'views/stores/QueryFiltersStore';
+import { QueryFiltersStore } from 'views/stores/QueryFiltersStore';
+import Query, { createElasticsearchQueryString, filtersForQuery, filtersToStreamSet } from 'views/logic/queries/Query';
+import type { FilterType, QueryId, TimeRange } from 'views/logic/queries/Query';
+import SearchBarForm from './searchbar/SearchBarForm';
 
 type Props = {
   availableStreams: Array<*>,
   config: any,
-  currentQuery: {
-    id: string,
-    timerange: any,
-    query: {
-      query_string: string,
-    },
-  },
+  currentQuery: Query,
   disableSearch: boolean,
-  onExecute: () => void,
-  queryFilters: Immutable.Map<any, any>,
+  onSubmit: ({ timerange: TimeRange, streams: Array<string>, queryString: string }, Query) => Promise<any>,
+  queryFilters: Immutable.Map<QueryId, FilterType>,
+};
+
+const defaultOnSubmit = ({ timerange, streams, queryString }, currentQuery: Query) => {
+  const newQuery = currentQuery.toBuilder()
+    .timerange(timerange)
+    .filter(filtersForQuery(streams))
+    .query(createElasticsearchQueryString(queryString))
+    .build();
+
+  return QueriesActions.update(newQuery.id, newQuery);
 };
 
-const SearchBar = ({ availableStreams, config, currentQuery, disableSearch = false, onExecute: performSearch, queryFilters }: Props) => {
+const SearchBar = ({ availableStreams, config, currentQuery, disableSearch = false, queryFilters, onSubmit = defaultOnSubmit }: Props) => {
```

Comment:
Should probably keep these to the `defaultProps` instead of potentially resetting what the default is here.

Reply:
The benefit of using default values is that flow is doing proper type checks, while the `PropTypes` type specs only do check very basic things (in this case if `onSubmit` is a function or not). Otoh, having runtime checks potentially helps us debugging issues caused by untested integration code.

The remaining risk is that someone could change one, but not the other, so they deviate. To address this, I created a `defaultProps` object which we are referencing in both places. I am not sure if it improves things or makes it more verbose.